drivers/serial/uart_stm32.c

```diff
@@ -873,7 +873,7 @@ static int uart_stm32_fifo_read_visitor(const struct device *dev, void *rx_data,
 {
```

Comment:
The typedef `fifo_read_fn` takes a `uint8_t` for third parameter, but that should also be changed by an `int`, that impacts the functions `fifo_read_with_u8` and `fifo_read_with_u16` as well.

Reply:
I have checked all codes related to the 15 serial drivers. There is only STM32's serial driver that has such impacts, while others are use the variable `num_tx` or `num_rx`  only as local variable, without transferring them to other codes.